dbt/adapters/snowflake/relation.py

```diff
@@ -106,3 +106,17 @@ def dynamic_table_config_changeset(
         if config_change_collection.has_changes:
             return config_change_collection
         return None
+
+    def as_case_sensitive(self) -> "SnowflakeRelation":
+        path_part_map = {}
+
+        for path in ComponentName:
+            if self.include_policy.get_part(path):
+                part = self.path.get_part(path)
+                if part:
+                    if self.quote_policy.get_part(path):
+                        path_part_map[path] = part
+                    else:
+                        path_part_map[path] = part.upper()
+
+        return self.replace_path(**path_part_map)
```

Comment:
I think this is the case, but to confirm, `replace_path` will return a copy of `self` with the new attribution, it won't modify the relation in place, correct?